### ISSUE TITLE
Model Serialization fails if DataNormalization supplied

### ISSUE THREAD

**Issue by Flips01** [deeplearning4j/deeplearning4j#7214]
Calling ModelSerializer.writeModel with a DataNormalization-Instance yields the following Exception:

> Exception in thread "main" java.io.IOException: flush() failed: stream is closed
> 	at org.apache.commons.io.output.ClosedOutputStream.flush(ClosedOutputStream.java:57)
> 	at org.apache.commons.io.output.ProxyOutputStream.flush(ProxyOutputStream.java:103)
> 	at java.base/java.util.zip.DeflaterOutputStream.flush(DeflaterOutputStream.java:283)
> 	at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:143)
> 	at java.base/java.io.FilterOutputStream.close(FilterOutputStream.java:182)
> 	at java.base/java.io.FilterOutputStream.close(FilterOutputStream.java:191)
> 	at org.deeplearning4j.util.ModelSerializer.writeModel(ModelSerializer.java:193)
> 	at org.deeplearning4j.util.ModelSerializer.writeModel(ModelSerializer.java:91)
> 	at TrainModel.main(TrainModel.java:116)
> 	Suppressed: java.io.IOException: flush() failed: stream is closed
> 		at org.apache.commons.io.output.ClosedOutputStream.flush(ClosedOutputStream.java:57)
> 		at org.apache.commons.io.output.ProxyOutputStream.flush(ProxyOutputStream.java:103)
> 		at java.base/java.util.zip.DeflaterOutputStream.flush(DeflaterOutputStream.java:283)
> 		at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:143)
> 		at java.base/java.io.DataOutputStream.flush(DataOutputStream.java:123)
> 		at java.base/java.io.FilterOutputStream.close(FilterOutputStream.java:182)
> 		... 3 more

A model file is still beeing generated. Calling ModelSerializer.writeModel without an DataNormalization instance doesn't result in an exception. 

This exception can be reproduced as follows:
[dl4j_exception.txt](https://github.com/deeplearning4j/deeplearning4j/files/2889762/dl4j_exception.txt)

**Comment by raver119** [comment]
Are you using latest dl4j/nd4j version?

**Comment by Flips01** [comment]
I am using 1.0.0-beta3. This should be the latest version shouldn't it?

**Comment by AlexDBlack** [comment]
Yes, 1.0.0-beta3 is the latest version.
This definitely looks like a bug. In the mean time, you can add the normalizer separately - i.e., writeModel to save the model, then addNormalizerToModel to update the file to include the normalizer.

https://github.com/deeplearning4j/deeplearning4j/blob/3c66853115e0d500cf1cafc0e060478766de4f03/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/util/ModelSerializer.java#L714-L722

**Comment by AlexDBlack** [comment]
I'm unable to reproduce this - I've tried 1.0.0-beta3 and current master/snapshots, Windows 10 and Kubuntu 17.10, with and without saving updater, running 1000x in a loop, trying both NormalizerStandardize and NormalizerMinMaxScaler, etc.

I've also looked at the ModelSerializer implementation and can't see how this bug could occur.

I'd be happy to take another look if I could reproduce the problem; I'll close this as not reproducible for now.
